code/src/components/Survey.js

```diff
@@ -0,0 +1,368 @@
+import React, { useState } from 'react';
+import { Header } from './Header'
+import { Button } from './Button';
+import { Dropdown } from './Dropdown';
+import { Radio } from './Radio';
+import { Text } from './Text';
+import { Checkbox } from './Checkbox';
+import {Progressbar } from './Progressbar';
+import { Summary } from './Summary';
+import './survey.css';
+
+export const Survey = () => {
+  const [question, setQuestion] = useState(-1);
+  const nextQuestion = () => setQuestion(question + 1);
+  const previousQuestion = () => setQuestion(question - 1);
+  const [ageGroup, setAgeGroup] = useState();
+  const [checkbox, setCheckbox] = useState(false); /* Needs to be specified to either true or false to be controlled from the start */
+  const [celebrationDrink, setCelebrationDrink] = useState('');
+  const [rainyFood, setRainyFood] = useState('');
+  const [restaurantHunt, setRestaurantHunt] = useState('');
+  const [saladMaker, setSaladMaker] = useState('');
+  const [userText, setUserText] = useState();
+  const [showSummary, setShowSummary] = useState(false);
+
+  /* This function uses a for-loop to extract the 'text' from each object inside the different 
+  arrays so that the text is displayed in summary and not the value.
+  The value could be used for backend instead (although at the moment it's just not used) */
+  const getUserInput = (userValue, optionsArray) => {
+    for (let i = 0; i < optionsArray.length; i++) {
+      if (optionsArray[i].value === userValue)
+        return optionsArray[i].text
+    } return 'no text found'
+  }
+  /* Arrays with objects for each components, needs to be definied outside of the component-scope so that getUserInput() can access it. */
```

Comment:
Maybe these arrays could be in their own components to clean up this file a bit

Reply:
If I want to move these into a different component, how shouldI specify which array I'd like to use in Survey.js and how can I make sure that getUserInput() can access them?

Reply:
You should just be able to import them as usual:

**drinkArray.js**
`export const drinkArray = [***]`

`import { drinkArray } from './drinkArray'`